### PR TITLE
Add algorithm-evaluating utility.

### PR DESCRIPTION
**Pull request by bobg**
Adapted in part from [rollsum_test.go](https://github.com/bobg/hashsplit/blob/17195adda444fcc11e96cfd6058613edd88af5be/rollsum_test.go) in github.com/bobg/hashsplit.

### CODE REVIEW

File: eval/eval.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+		correlations [32 * 32]int
+	)
+
+	r := factory()
+	var elapsed time.Duration
+	checkpoint := time.Now()
+	for i := 0; i < len(megabyte); i++ {
+		r.Roll(megabyte[i])
+		digest := r.Digest()
+		elapsed += time.Since(checkpoint) // stop timer
```
Comment (zenhack) [inline eval/eval.go]: We should probably just measure the running time in a separate pass from the statistical bookkeeping entirely; my sense is that time.Now() on most platforms is probably noticably *more* expensive than the checksum operations, as it requires a system call (though Linux may be an exception here). So I don't trust this not to dilute the differences between hash functions. The results I'm seeing running on my laptop argue for performance not being that different across hashes, but I have a sinking feeling that we may just be measuring time.Now().

**Review by zenhack — CHANGES_REQUESTED** [review]
Thanks for putting this together! I have one concern inline re: timing measurements, but this looks useful.

File: eval/eval.go (hunk 1 above)
Comment (bobg) [reply eval/eval.go]: Quite right, thanks! Fixed.

**Comment by zenhack** [comment]
Merged, thanks!

**Pull request merged**